pkg/machine/provider/platform_darwin.go

```diff
@@ -36,3 +41,45 @@ func Get() (vmconfigs.VMProvider, error) {
 		return nil, fmt.Errorf("unsupported virtualization provider: `%s`", resolvedVMType.String())
 	}
 }
+
+// SupportedProviders returns the providers that are supported on the host operating system
+func SupportedProviders() []define.VMType {
+	return []define.VMType{define.AppleHvVirt}
+}
+
+// InstalledProviders returns the supported providers that are installed on the host
+func InstalledProviders() ([]define.VMType, error) {
+	var outBuf bytes.Buffer
+	// Apple's Virtualization.Framework is only supported on MacOS 11.0+
+	const SupportedMacOSVersion = 11
```

Comment:
We might need to bump this to version 13 instead. 
https://github.com/containers/podman/issues/22121#issuecomment-2012729242

Reply:
Other than this though, LGTM.

Reply:
Yes, podman machine's use of vfkit+uefi requires macos 13